src/contract_wrappers/modules/checkpoint/dividend_checkpoint_wrapper.ts

```diff
@@ -171,12 +180,12 @@ export default abstract class DividendCheckpointWrapper extends ModuleWrapper {
       created: bigNumberToDate(result[1]),
       maturity: bigNumberToDate(result[2]),
       expiry: bigNumberToDate(result[3]),
-      amount: result[4],
-      claimedAmount: result[5],
-      totalSupply: result[6],
+      amount: weiToValue(result[4], await (await this.securityTokenContract()).decimals.callAsync()),
+      claimedAmount: weiToValue(result[5], await (await this.securityTokenContract()).decimals.callAsync()),
+      totalSupply: weiToValue(result[6], await (await this.securityTokenContract()).decimals.callAsync()),
```

Comment:
You should call `await (await this.securityTokenContract()).decimals.callAsync())` at the beginning and store the value in a variable to avoid to call it three times.